uberfire-security/uberfire-security-management/uberfire-security-management-backend/src/main/java/org/uberfire/ext/security/management/service/GroupsManagerServiceImpl.java

```diff
@@ -0,0 +1,110 @@
+import org.uberfire.ext.security.management.api.exception.NoImplementationAvailableException;
+import org.uberfire.ext.security.management.api.service.GroupsManagerService;
+import org.uberfire.ext.security.management.util.SecurityManagementUtils;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * <p>The UberFire service implementation for GroupsManager API.</p>
+ */
+@Service
+@ApplicationScoped
+public class GroupsManagerServiceImpl implements GroupsManagerService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GroupsManagerServiceImpl.class);
+    
+    @Inject
+    private BackendUserSystemManager userSystemManager;
+    
+    private GroupsManager service;
+    
+    @PostConstruct
+    public void init() {
+        service = userSystemManager.groups();
+    }
+    
+    private GroupsManager getService() throws GenericUsersSystemManagementException {
+        if (service == null) throw new NoImplementationAvailableException();
+        return service;
+    }
+
+    @Override
+    public SearchResponse search(SearchRequest request) throws GenericUsersSystemManagementException {
+        final GroupsManager serviceImpl = getService();
+        if (request.getPage() == 0) throw new IllegalArgumentException("First page must be 1.");
+        return serviceImpl.search(request);
+    }
+
+    @Override
+    public Group get(String identifier) throws GenericUsersSystemManagementException {
+        final GroupsManager serviceImpl = getService();
+        return serviceImpl.get(identifier);
+    }
+
+    @Override
+    public Group create(Group group) throws GenericUsersSystemManagementException {
+        final GroupsManager serviceImpl = getService();
+        return serviceImpl.create(group);
+    }
+
+    @Override
+    public Group update(Group group) throws GenericUsersSystemManagementException {
+        final GroupsManager serviceImpl = getService();
+        return serviceImpl.update(group);
+    }
+
+    @Override
+    public void delete(String... identifiers) throws GenericUsersSystemManagementException {
+        final GroupsManager serviceImpl = getService();
+        serviceImpl.delete(identifiers);
+    }
+
+    @Override
+    public void assignUsers(String name, Collection<String> users) throws GenericUsersSystemManagementException {
+        final GroupsManager serviceImpl = getService();
+        serviceImpl.assignUsers(name, users);
+    }
+
+    @Override
+    public CapabilityStatus getCapabilityStatus(Capability capability) {
+        final GroupsManager serviceImpl = getService();
+        return serviceImpl.getCapabilityStatus(capability);
+    }
+
+    private void handleGenericException(RuntimeException e) throws GenericUsersSystemManagementException {
```

Comment:
We've already discusses this but for completeness. It's not clear what this method does. It shouldn't be needed (see below).